### PR TITLE
Value profiling API changes

### PR DESCRIPTION
**Pull request by ncough**
A new value profiling implementation, JProfiling, is being
added to OpenJ9, along with some changes to the value
profiling API. These change updates these calls as necessary
and adds configuration options for the profiling modes.

### CODE REVIEW

**Comment by charliegracie** [comment]
genie-omr build all

**Comment by andrewcraik** [comment]
This looks like general clean-up and adding a few options and hook points. LGTM from me pending any further changes.

File: compiler/optimizer/Inliner.hpp [hunk 1]
```diff
@@ -72,7 +72,6 @@ const float SECOND_BEST_MIN_CALL_FREQUENCY = .2275f; //65% of 35%
 
 class TR_BitVector;
 class TR_CallStack;
-class TR_ExtraAddressInfo;
```
Comment (vijaysun-omr) [inline compiler/optimizer/Inliner.hpp]: We seem to be removing TR_ExtraAddressInfo and using TR_ExtraValueInfo instead of TR_ExtraAbstractInfo. I assume these two changes are related in that TR_ExtraAddressInfo is the only other subclass of TR_ExtraAbstractInfo. Could you please elaborate on why this change is being made ?

Comment (ncough) [reply compiler/optimizer/Inliner.hpp]: `TR_ExtraAddressInfo` has been removed here as it wasn't being used and its a small pain to forward declare, as its now a template. The other changes from `TR_ExtraAbstractInfo` to `TR_ExtraValueInfo` are due to the removal of `TR_ExtraAbstractInfo`. Rather than having to cast back and forth between the two classes, `TR_ValueInfo`'s methods now use `TR_ExtraValueInfo` directly.

File: compiler/control/OMROptions.cpp [hunk 2]
```diff
@@ -4797,6 +4798,7 @@ char *OMR::Options::_verboseOptionNames[TR_NumVerboseOptions] =
    "hookDetailsClassLoading",
    "hookDetailsClassUnloading",
    "sampleDensity",
+   "profiling"
```
[the remark this answers is not in the document]
Comment (ncough) [reply compiler/control/OMROptions.cpp]: For verbose options, the option name has to match its position in `TR_VerboseFlags`, rather than being alphabetically sorted like the main option list. Here is matches the value for `TR_VerboseProfiling`.
